### ISSUE TITLE
Code the architectures

### ISSUE THREAD

**Issue by nolan-cerebras** [ndey96/deep-generator-network#1]
- [x] Code the encoder
- [x] Code the decoder
- [x] Code the generator
- [x] Code the discriminator

**Comment by nolan-cerebras** [comment]
[unofficial tensorflow implementation](https://github.com/shijx12/DeepSim) uses batchnorm however the original paper makes no mention of it

**Comment by nolan-cerebras** [comment]
original implementation: https://lmb.informatik.uni-freiburg.de/resources/binaries/arxiv2016_alexnet_inversion_with_gans/release_deepsim_v0.5.zip

llink to list of papers from authors' group: https://lmb.informatik.uni-freiburg.de/resources/software.php

**Comment by nolan-cerebras** [comment]
the paper deepsim paper says their upconvolutional layers consist of bed of nails upsampling, followed by regular convolution. Upon looking into their original code, they are actually using transposed convolution 😮

**Comment by nolan-cerebras** [comment]
Done with 4e615ce8b546714110c9beb709b95bac5bc92103
